teps/0122-complete-build-instructions-and-parameters.md

```diff
@@ -0,0 +1,241 @@
+
+### TaskRun Specification
+
+The table below provides the name and description of the API fields that the TaskRun requires. In addition, it also lists if that information is required, not required, or provided by the provenance. If it is required, it also shows where the information should be provided. Note that since task run provides information during runtime, most of the information is contained in the **invocation.Parameters** section of the provenance.
+
+
+| | **Metadata**      | | |
+| ------ | ------------- | ----- | ----- |
+| **Field Name**    | **Description** | **Not Required / Required / Provided by provenance** | **Insert in provenance** |
+| name| Name of the task run| Required (provide an audit trail)|  |
+| | **Spec** | | |
+| resources | resources used by task | Provided (can be extracted from steps/entrypoint) |  |
+| service account name | name of the service account | Required | buildConfig |
+| params | parameter values provided by taskrun | Provided |  |
+| workspaces | workspaces used by the task | Required | buildConfig |
+| pod template | | Required | buildConfig |
+| Timeout | time in which a task should complete | Not Required (can probably use default?) |  |
+| StepOverrides | Override Step configuration specified in a Task. Currently we can override compute resources. | Not Required |  |
+| SidecarOverrides | Override Sidecar configuration specified in a Task. Currently we can override compute resources. | Not Required |  |
+| ComputeResources | Configure compute resources required by the steps in the task. | Required (because this could also indicate a minimum amount of resources required for the task run.) | buildConfig |
+| [TaskSpec](#task-specification) | Specification of the resolved task | See [spec](#task-specification) | buildConfig |
+| [TaskRef](#taskref)  | Details of the referenced task | Not Required (Since we save the complete TaskSpec even that of a remote task) |  |
+| | **Status** | | |
+| Task Results | Results produced by the task run | Not Required | buildConfig |
+
+
+### Configuration Feature Flags
+
+The feature flags that a user/operator specified during installation of Tekton pipelines that lead to the build are also required. These can be added to the **invocation.Environment** section of the [provenance](#provenance-for-executed-taskrun). The feature flags should be fetched by the task run reconciler and embedded in the **task run status** so that chains can access it easily.
```

Comment:
What about flags passed to the controller directly at start time?
Also, we have a lot of different configuration flags, so it would make sense to specify which ones shall be included.

It feels like this would cause a non-negligable increase in size of the `TaskRun` resources. 
Is it a standard SLSA requirement to include the configuration of the build system?

Reply:
>What about flags passed to the controller directly at start time?
>
Hmm. I didn't think of that. May be we can address that in a later PR.

>Also, we have a lot of different configuration flags, so it would make sense to specify which ones shall be included.
I was thinking of including them all since there are only [16](https://github.com/tektoncd/pipeline/blob/8c59d00342e9479e68e9c52f6d18b62ac3015489/pkg/apis/config/feature_flags.go#L97-L112) I think. 
>It feels like this would cause a non-negligable increase in size of the TaskRun resources.
Hmm. Unless I'm misunderstanding, the only features that I'm thinking of are [these](https://github.com/tektoncd/pipeline/blob/8c59d00342e9479e68e9c52f6d18b62ac3015489/pkg/apis/config/feature_flags.go#L97-L112). That shouldn't be an issue right?
We can definitely update this approach if its not good enough. I've linked you to the original discussion with Billy as to why it would be nice to have the configuration information in the task run status. 
>Is it a standard SLSA requirement to include the configuration of the build system?
>
I think SLSA wants reproducibility as best as we can. I think feature flags that were used to configure Tekton that produced the build is very important for reproducibility.